Automattic/wp-calypso
# Store Promotions: Add appliesTo component

This adds a tree selector component for "Applies To" on the form for
promotion editing. It allows selection of all products, product
categories, or individual products.

To Test:
1. `npm start`
2. Visit `http://calypso.localhost:3000/store/promotion/<site url>`
3. Observe "Applies To" selector, it should load with products and product categories.
4. Click a product, observe it being selected.
5. Click that product's category, observe all products selected.
6. Click the first product again, observe that it does not unselect because its category is selected.
7. Click the product category again, all should be unselected now.
8. Attempt above sequence (4-7) with the "All" selection.
9. If a product has more than one category, it should always show as selected/unselected in both places.
